### PR TITLE
Fix: markdown issue and allow app:// deeplinks

### PR DESCRIPTION
**Pull request by jaapbakker88**
# Submit a pull request

* fixes an issue when typing markdown manually it also tries to convert the link to markdown
* allows app:// deeplinks

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -149,6 +149,25 @@ const allowedMarkups = [
   'delete',
 ];
 
+const matchMarkdownLinks = (message) => {
+  const regexMdLinks = /\[([^\[]+)\](\(.*\))/gm;
+  const matches = message.match(regexMdLinks);
+  // console.log('links', matches);
+
+  const singleMatch = /\[([^\[]+)\]\((.*)\)/;
+  var links = [];
```
Comment (vicnicius) [inline src/utils.js]:
```suggestion
  let links = [];
```

File: src/utils.js [hunk 2]
```diff
@@ -149,6 +149,25 @@ const allowedMarkups = [
...
+  if (matches) {
+    for (var i = 0; i < matches.length; i++) {
```
Comment (vicnicius) [inline src/utils.js:161]:
```suggestion
    for (let i = 0; i < matches.length; i++) {
```

File: src/utils.js [hunk 3]
```diff
@@ -149,6 +149,25 @@ const allowedMarkups = [
...
+      var text = singleMatch.exec(matches[i]);
```
Comment (vicnicius) [inline src/utils.js:162]:
```suggestion
      let text = singleMatch.exec(matches[i]);
```

File: src/utils.js [hunk 4]
```diff
@@ -149,23 +150,48 @@ const allowedMarkups = [
   'delete',
 ];
 
+const matchMarkdownLinks = (message) => {
+  const regexMdLinks = /\[([^\[]+)\](\(.*\))/gm;
+  const matches = message.match(regexMdLinks);
+  // console.log('links', matches);
+
+  const singleMatch = /\[([^\[]+)\]\((.*)\)/;
+  let links = [];
```
Comment (vicnicius) [inline src/utils.js:159]: Minor, but we could go functional and replace the `for` loop for `matches.map(match => singleMatch.exec(match)[2]);` (didn't test) here.

Comment (jaapbakker88) [reply src/utils.js]: good one, pushed that.

**Pull request merged**